mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -416,7 +416,13 @@ final class OneVsRest @Since("1.4.0") (
       multiclassLabeled.persist(StorageLevel.MEMORY_AND_DISK)
     }
 
-    val executionContext = getExecutionContext
+    val executionContext = getParallelism match {
+      case 1 =>
+        ThreadUtils.sameThread
+      case n =>
+        ExecutionContext.fromExecutorService(ThreadUtils
+          .newDaemonCachedThreadPool(s"${this.getClass.getSimpleName}-thread-pool", n))
+    }
```

Comment:
This part code is used in several places, we should define a helper function for it.

Reply:
Done, add a new trait `HasExecutionContext  extends HasParallelism` for it